src/palace/manager/api/controller/circulation_manager.py

```diff
@@ -147,9 +149,13 @@ def load_work(
         if isinstance(pools, ProblemDetail):
             return pools
 
-        # We know there is at least one LicensePool, and all LicensePools
-        # for an Identifier have the same Work.
-        work: Work = pools[0].work
+        # We know there is at least one LicensePool. Find the first one with
+        # a work set on it.
+        work: Work | None = first_or_default([lp.work for lp in pools if lp.work])
```

Comment:
This is where the problem we were seeing in production stems from. `pools[0].work` was `None`, and that's where the trouble began. In that case there was another license pool that had a work set, so we try to find the first LP with a work set instead of just using the first one.

I also added some handling for the case where there is no work, since this case is still possible.